### PR TITLE
Enabling users to upload custom GeoJSON

### PR DESCRIPTION
**Pull request by Shivamdhar**
### Description

- Enables users to upload custom GeoJSON files 
- Adds UI elements, API layer and backend connectivity code with OpenSearch cluster and geospatial plugin
 
<table>
  <tr>
    <td></td>
    <td></td>
  </tr>
 </table>

### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1408 
Task - Allow users to upload custom vector map
 
### Check List
- [x] New functionality includes testing.
        
       
           yarn test:jest src/plugins/region_map/*                                  
           yarn run v1.22.18
           $ node scripts/jest src/plugins/region_map/common src/plugins/region_map/config.ts 
           src/plugins/region_map/opensearch_dashboards.json src/plugins/region_map/package.json 
           src/plugins/region_map/public src/plugins/region_map/server src/plugins/region_map/target
           PASS  src/plugins/region_map/public/region_map_fn.test.js
           PASS  src/plugins/region_map/public/components/empty_prompt.test.tsx
           PASS  src/plugins/region_map/public/components/vector_upload_options.test.tsx
           PASS  src/plugins/region_map/public/components/custom_vector_upload.test.tsx
          
          Test Suites: 4 passed, 4 total
          Tests:       16 passed, 16 total
          Snapshots:   5 passed, 5 total
          Time:        10.499 s

        
  - [ ] All tests pass
    - [x] `yarn test:jest`
    - [x] `yarn test:jest_integration`
       ```
        Test Suites: 50 passed, 50 total
        Tests:       5 skipped, 438 passed, 443 total
        Snapshots:   77 passed, 77 total
        Time:        329.189 s
        Ran all test suites.
                 │ info [opensearch] cleanup complete
        Done in 332.29s.
       ```
    - [ ] `yarn test:ftr`
      
- [x] New functionality has been documented.
   RFC can be accessed here: https://github.com/opensearch-project/geospatial/issues/47

- [x] Commits are signed per the DCO using --signoff 

### Todo:

- Add unit tests

### CODE REVIEW

File: src/plugins/region_map/server/services/utils/constants.ts
Comment (junqiu-lei) [inline src/plugins/region_map/server/services/utils/constants.ts:8]:
```suggestion
  'Accept': 'application/json',
```

File: src/plugins/region_map/public/components/custom_vector_upload.tsx
Comment (junqiu-lei) [inline src/plugins/region_map/public/components/custom_vector_upload.tsx:78]: How about add to new line for each error?

Comment (Shivamdhar) [reply src/plugins/region_map/public/components/custom_vector_upload.tsx:78]: yup will add that!

File: src/plugins/region_map/server/services/utils/constants.ts
Comment (Shivamdhar) [reply src/plugins/region_map/server/services/utils/constants.ts:8]: Had added quotes, the lint checker removed it : )

File: config/opensearch_dashboards.yml
Comment (VijayanB) [inline config/opensearch_dashboards.yml:21]: As discussed offline, let's not update this here, instead, mention this in documentation.

File: src/plugins/region_map/public/components/custom_vector_upload.tsx
Comment (VijayanB) [reply src/plugins/region_map/public/components/custom_vector_upload.tsx:78]: shall we use constant in message as well instead of hard coding the number?

File: src/plugins/region_map/public/components/custom_vector_upload.tsx
Comment (VijayanB) [inline src/plugins/region_map/public/components/custom_vector_upload.tsx:111]:
```suggestion
    } 
    errorIndexNameDiv.textContent = '';
    return true;
    }
```

File: src/plugins/region_map/public/components/custom_vector_upload.tsx
Comment (VijayanB) [inline src/plugins/region_map/public/components/custom_vector_upload.tsx:117]: Constant for limit. Hence, if we change it future, we don't have to update here

File: src/plugins/region_map/server/services/opensearch_service.js
Comment (VijayanB) [inline src/plugins/region_map/server/services/opensearch_service.js:8]: nit: driver

File: src/plugins/region_map/server/clusters/geo_cluster.js
Comment (VijayanB) [inline src/plugins/region_map/server/clusters/geo_cluster.js:6]: shall we use geospatial instead of geo?

File: src/plugins/region_map/server/services/utils/constants.ts
Comment (VijayanB) [inline src/plugins/region_map/server/services/utils/constants.ts:15]: Where is ADMIN const is used?

**Comment by VijayanB** [comment]
@kavilla From our POV this PR looks good other than some minor feedback. Can you please take a look at it? Thanks.

File: src/plugins/region_map/server/services/utils/constants.ts
Comment (Shivamdhar) [reply src/plugins/region_map/server/services/utils/constants.ts:15]: that was for some test, will remove the unused constants from here.

File: src/plugins/region_map/public/components/custom_vector_upload.tsx
Comment (Shivamdhar) [reply src/plugins/region_map/public/components/custom_vector_upload.tsx:117]: Adding a constant. Will change it to error as well.

File: config/opensearch_dashboards.yml
Comment (Shivamdhar) [reply config/opensearch_dashboards.yml:21]: Changed this to route specific configuration in `server/routes/geospatial.ts`.

```
options: {
        body: {
          accepts: 'application/json',
          maxBytes: FILE_PAYLOAD_SIZE, // 25MB payload limit
        },
      }
```

Thanks @vamshin @VijayanB

**Comment by tmarkley** [comment]
@VijayanB @Shivamdhar the core Dashboards team has expressed some concerns about this approach in the original issue: #1408. Can we discuss the open questions first before proceeding?

**Comment by Shivamdhar** [comment]
> @VijayanB @Shivamdhar the core Dashboards team has expressed some concerns about this approach in the original issue: #1408. Can we discuss the open questions first before proceeding?

Sure Tommy, will jot down the thoughts in that thread by eod. Thanks for taking out time : )

**Comment by tmarkley** [comment]
Excluding the open conversation about the approach itself, we need test coverage before this is merged. There are no new tests here. If you need help determining how to implement tests, please let us know.

**Comment by Shivamdhar** [comment]
> Excluding the open conversation about the approach itself, we need test coverage before this is merged. There are no new tests here. If you need help determining how to implement tests, please let us know.

Yes, I'm on it. I'll be adding tests to the PR as I'm getting to learn the different testing strategies on this side as well. Currently, working on adding snapshot tests. Will keep the PR updates with the test cases as and when I write them.

**Review by tmarkley — CHANGES_REQUESTED** [review]
Blocking this until tests are added. Take a look at our guide for [Writing Tests](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/TESTING.md#writing-tests) and feel free to reach out if you need additional help!

**Comment by Shivamdhar** [comment]
We have decided to move the component (Import custom map tab) out from region map and make it part of the `dashboards-maps` plugin. Closing this PR in favor of https://github.com/opensearch-project/dashboards-maps/pull/7.

Thank you all for you suggestions and guidance.

**Pull request closed without merging**
